### PR TITLE
#3694 Select a vaccinator

### LINKED ISSUE #3694: UI For adding a vaccinator to a vaccine prescription
## Is your feature request related to a problem? Please describe.

A drop down on step 3 of a vaccine prescription to choose a vaccinator.

## Describe the solution you'd like

A drop down on step 3 of a vaccine prescription to choose a vaccinator.

## Implementation

A drop down on step 3 of a vaccine prescription to choose a vaccinator.

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3694 

## Change summary

- INCLUDES Code from #3693 and #3690
- Adds ability to select a vaccinator. Defaults to the most recent vaccinator selected.

## Testing

- [ ] Make a vaccine prescription , can select a vaccinator

### Related areas to think about
(My vaccineator has no code/first/last)
image

### CODE REVIEW

File: src/widgets/Tabs/VaccineSelect.js [hunk 1]
```diff
@@ -136,46 +168,70 @@ const VaccineSelectComponent = ({
...
   const selectedRows = selectSelectedRows(state);
   const selectedBatchRows = selectSelectedBatchRows(state);
   const selectedBatches = selectSelectedBatches(state);
   const selectedVaccines = selectSelectedVaccines(state);
   const vaccines = selectVaccines(state);
   const [selectedVaccine] = selectedVaccines;
 
-  return { selectedBatches, selectedBatchRows, selectedRows, selectedVaccine, vaccines };
+  return {
+    vaccinator,
```
Comment (mark-prins) [inline src/widgets/Tabs/VaccineSelect.js:203]: the `vaccinator` isn't set in the method - is there a commit missing?

Comment (joshxg) [reply src/widgets/Tabs/VaccineSelect.js:203]: Ah yeah I think there is a commit missing which will mess the next PR up also 😡

Comment (joshxg) [reply src/widgets/Tabs/VaccineSelect.js:205]: f8a8e47

sorry!

**Comment by mark-prins** [comment]
Just looking into why I'm not getting the vaccinator saving.. 

Screen Shot 2021-03-09 at 10 11 25 AM

File: src/database/DataTypes/TransactionItem.js [hunk 2]
```diff
@@ -286,6 +286,15 @@ export class TransactionItem extends Realm.Object {
     this.setTotalQuantity(database, newTotalQuantity);
   }
 
+  setVaccinator(database, vaccinator) {
+    this.batches.forEach(batch =>
+      database.update('TransactionBatch', {
+        id: batch.id,
+        vaccinator,
```
Comment (mark-prins) [inline src/database/DataTypes/TransactionItem.js:293]:
```suggestion
        medicineAdministrator: vaccinator,
```

File: src/reducers/Entities/VaccinePrescriptionReducer.js [hunk 3]
```diff
@@ -48,6 +50,31 @@ export const VaccinePrescriptionReducer = (state = initialState(), action) => {
 
       return { ...state, selectedBatches: [itemBatch] };
     }
+
+    case VACCINE_PRESCRIPTION_ACTIONS.UPDATE: {
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/reducers/Entities/VaccinePrescriptionReducer.js]: 1b77ca8

File: src/database/DataTypes/MedicineAdministrator.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Sustainable Solutions (NZ) Ltd. 2021
+ * mSupply Mobile
+ */
+
+import Realm from 'realm';
+
+export class MedicineAdministrator extends Realm.Object {
+  get displayString() {
+    return `${this.code ?? ''} ${this.firstName ?? ''}${this.lastName ?? ''}`.trim();
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/database/DataTypes/MedicineAdministrator.js]: abe40fa

File: src/database/DataTypes/TransactionItem.js (hunk 2 above)
Comment (joshxg) [reply src/database/DataTypes/TransactionItem.js]: 35eae88

**Review by mark-prins — APPROVED** [review]
LGTM! thanks - works nicely :)

**Pull request merged**